### DISCUSSION TITLE
Janusgraph not picking up Index during lookup.

### DISCUSSION THREAD

**Discussion by kaintharinder** [JanusGraph/janusgraph#2992] [General]
Hi ,

Below is my Schema Model

<img width="705" alt="Screenshot 2022-03-23 at 10 48 09 PM" src="https://user-images.githubusercontent.com/56577372/159758203-daa79dfa-32e0-4b7b-a688-992a43771cc9.png">

I have created one composite index as well on docId and inserted some 100 records of vertex Segment Type

**Query** : g.V().has('docId','1d5bdb14-09a1-4259-a448-b8fc59ce14d0') 

This prints the correct results, but in the janusgraph logs I see a warning that query is scanning all vertices, use Index.
Seems like my index is not getting picked up.
The profile command shows it is doing full scan and not using the index.
<img width="797" alt="Screenshot 2022-03-23 at 10 51 22 PM" src="https://user-images.githubusercontent.com/56577372/159758710-945b0aa9-32fc-489d-ae63-ceee6346f6b2.png">

I am using cassandra as storage backend. **Is there anything that I am missing**. In the Schema screenshot attached , I see the index as installed.

**Comment by vtslab** [comment]
Your printed schema says that the byDocId index has status INSTALLED. Please check [https://docs.janusgraph.org/schema/index-management/index-lifecycle/](https://docs.janusgraph.org/schema/index-management/index-lifecycle/) and see if that helps.

**Comment by kaintharinder** [comment]
Thanks @vtslab  .. The lifecycle helped me. Apparently there was some problem in which the index was not getting enabled.

**Comment by kaintharinder** [comment]
I have one more doubt. My graph is as below
![graph](https://user-images.githubusercontent.com/56577372/159938869-8c79b588-ba45-4737-b935-fb8e738f031f.jpg)

Page ----child---> Instance
Instance -----> targets -----> Segment (can be more than 1)
instance -----> emdeds -----> Fragment (can be more than 1)
Fragment -----> child -------> FragmentInstance 
Fragment -----> targets -----> Segment (Can be more than 1)

The query will start from Page (V1) and need to return the full tree such as 
v1,v2,v3,v4,v5,v6,v7  // their ids will suffice.

**Query** : g.V().has('docId','page1').emit().repeat(out()).tree()

This prints all vertices but it prints duplicates as well in case the segment is shared between instance and fragment Instance.
Another thing is this query is very slow for higher number of vertices.

I have one index for lookup by docId

**Comment by vtslab** [comment]
You can probably find inspiration for improvement at: 
[https://tinkerpop.apache.org/docs/current/recipes/#connected-components](https://tinkerpop.apache.org/docs/current/recipes/#connected-components)
